File: 2022/C++/Day-1/Part-1.cpp
#include <iostream>
#include <fstream>
#include <string>

int main() {
    std::ios_base::sync_with_stdio(false);
    std::ifstream inputFile("input.txt");
    //-------------------------------------
    int max_sum(0);
    
    std::string line;
    int sum(0);
    while (std::getline(inputFile, line)) {
        if (!line.empty()) {
            sum += std::stoi(line);
        } else {
            max_sum = std::max(max_sum, sum);
            sum = 0;
        }
    }
    max_sum = std::max(max_sum, sum);
    

    std::cout << max_sum << "\n";
    //-------------------------------------
    return 0;
}

File: 2022/Python/Day-1/Part-1.py
from helper_functions import get_relative_path


def main():
    with open(get_relative_path("input.txt")) as file:
        lines = file.read().split("\n\n")
        sums = [sum(map(int, line.split("\n"))) for line in lines]
    print(max(sums))


if __name__ == "__main__":
    main()
